### PR TITLE
Added updated ZeroMQ 4.2.3 (which is also compatible with Windows UWP)

### PR DESCRIPTION
**Pull request by fwinnen**

### CODE REVIEW

**Comment by fwinnen** [comment]
Yeah, there are braking API changes in zmq::poller, this is why i did not upgrade the default one, to not break existing code

**Comment by ruslo** [comment]
Add this note to documentation and update example, otherwise it make no sense to merge this change.

**Comment by fwinnen** [comment]
Done!
Is the doc enough for you?

File: docs/packages/pkg/ZeroMQ.rst [hunk 1]
```diff
@@ -30,7 +30,18 @@ ZeroMQ
...
 
+Versions
+--------
+
+Hunter provides two separate versions of ZeroMQ:
+
+-  ZeroMQ 4.1.4 (default)
+-  ZeroMQ 4.2.3
+
+Since 4.2.3 introduces some API-breaking changes, 4.1.4 remains the default version for hunter.
```
Comment (ruslo) [inline docs/packages/pkg/ZeroMQ.rst]: > API-breaking changes

C++ API?

File: examples/ZeroMQ/config.cmake [hunk 2]
```diff
@@ -0,0 +1,4 @@
+hunter_config(ZeroMQ
+    VERSION 4.2.3
```
Comment (ruslo) [inline examples/ZeroMQ/config.cmake]: We will test `4.2.3` but `4.1` remains default? I don't think it's good idea.

**Comment by fwinnen** [comment]
Ok, changed to latest as default.
4.2.2 is the official stable version and this 4.2.3 adds some fixes needed for windows universal.

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2432
* https://travis-ci.org/ingenue/hunter/builds/277368422?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2436
* https://travis-ci.org/ingenue/hunter/builds/277669633?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.19.108

**Pull request merged**